test/conformance/serial/ptp.go

```diff
@@ -838,6 +840,260 @@ var _ = Describe("["+strings.ToLower(DesiredMode.String())+"-serial]", Serial, f
+						return buf.String()
+					}, pkg.TimeoutIn5Minutes, 5*time.Second).Should(ContainSubstring("phc2sys"),
+						"phc2ys process status not detected")
+
+					time.Sleep(10 * time.Second)
+					buf, _, _ := pods.ExecCommand(client.Client, fullConfig.DiscoveredClockUnderTestPod, pkg.PtpContainerName, []string{"curl", pkg.MetricsEndPoint})
+					ret, err := processRunning(buf.String())
+					Expect(err).To(BeNil())
+					Expect(ret["phc2sys"]).To(BeTrue(), "Expected phc2sys to be running for GM")
+					Expect(ret["gpspipe"]).To(BeTrue(), "Expected gpspipe to be running for GM")
+					Expect(ret["ts2phc"]).To(BeTrue(), "Expected ts2phc to be running for GM")
+					Expect(ret["gpsd"]).To(BeTrue(), "Expected gpsd to be running for GM")
+					Expect(ret["ptp4l"]).To(BeTrue(), "Expected ptp4l to be running for GM")
+					time.Sleep(1 * time.Minute)
+				})
+
+				By("checking clock class state is locked", func() {
+					/*
+						# TYPE openshift_ptp_clock_class gauge
+						# openshift_ptp_clock_class{node="cnfdg32.ptp.eng.rdu2.dc.redhat.com",process="ptp4l"} 6
+					*/
+					Eventually(func() string {
+						buf, _, _ := pods.ExecCommand(client.Client, fullConfig.DiscoveredClockUnderTestPod, pkg.PtpContainerName, []string{"curl", pkg.MetricsEndPoint})
+						return buf.String()
+					}, pkg.TimeoutIn5Minutes, 5*time.Second).Should(ContainSubstring(metrics.OpenshiftPtpClockClass),
+						"NMEA status metrics are not detected")
+					buf, _, _ := pods.ExecCommand(client.Client, fullConfig.DiscoveredClockUnderTestPod, pkg.PtpContainerName, []string{"curl", pkg.MetricsEndPoint})
+					clockClassPattern := `openshift_ptp_clock_class\{node="([^"]+)",process="([^"]+)"\} (\d+)`
+
+					// Compile the regular expression
+					clockClassRe := regexp.MustCompile(clockClassPattern)
+
+					// Find matches
+					clockClassMap := map[string]bool{"ptp4l": false}
+
+					scanner := bufio.NewScanner(strings.NewReader(buf.String()))
+					timeout := 1 * time.Minute
+					start := time.Now()
+					for scanner.Scan() {
+						t := time.Now()
+						elapsed := t.Sub(start)
+						if elapsed > timeout {
+							fmt.Println("Timedout reading input from metrics")
+						}
+						line := scanner.Text()
+						if matches := clockClassRe.FindStringSubmatch(line); matches != nil {
+							if _, ok := clockClassMap[matches[2]]; ok && matches[3] == "6" {
+								clockClassMap[matches[2]] = true
+								break
+							}
+
+						}
+					}
+					if err := scanner.Err(); err != nil {
+						fmt.Println("Error reading input:", err)
+					}
+					Expect(clockClassMap["ptp4l"]).To(BeTrue(), "Expected ptp4l clock class state to be Locked for GM")
+				})
+
+				By("checking DPLL frequency state locked", func() {
+					/*
+						# TYPE openshift_ptp_frequency_status gauge
+						# openshift_ptp_frequency_status{from="dpll",iface="ens7fx",node="cnfdg32.ptp.eng.rdu2.dc.redhat.com",process="dpll"} 3
```

Comment:
for multiple cards, each card could have one line. Need to check all, or add a TODO here.
```
openshift_ptp_frequency_status{from="dpll",iface="ens3fx",node="cnfdg4.ptp.eng.rdu2.dc.redhat.com",process="dpll"} 3
openshift_ptp_frequency_status{from="dpll",iface="ens4fx",node="cnfdg4.ptp.eng.rdu2.dc.redhat.com",process="dpll"} 3
```

Reply:
added a TODO